### PR TITLE
Gleefre/various fixes

### PR DESCRIPTION
**Pull request by Gleefre**
Reopening #77 with additional changes.

Please see commit messages.

### CODE REVIEW

File: src/utils.lisp [hunk 1]
```diff
@@ -10,11 +10,12 @@
 
 (defparameter *build* nil)
 
-(defun pad-list (list pad length)
-  (if (>= (length list) length)
-      list
-      (append (make-list (- length (length list)) :initial-element pad)
-              list)))
+(defun pad-list (list pad length
+                 &aux (pad-length (- length (length list))))
```
Comment (vydd) [inline src/utils.lisp:14]: nit/personal prefs: I don't think &aux makes things more readable, so I'll change it to a let in a followup

Comment (Gleefre) [reply src/utils.lisp:14]: Fine by me.

File: src/utils.lisp [hunk 2]
```diff
@@ -34,11 +34,10 @@
        finally (return (nreverse acc)))))
 
 (defun group-bits (x &optional (bits 8))
-  (let ((bit-fill (1- (expt 2 bits))))
-    (do* ((x x (ash x (- bits)))
-          (acc `(,(boole boole-and x bit-fill))
-               (cons (boole boole-and x bit-fill) acc)))
-         ((zerop x) (cdr acc)))))
+  (loop with result = ()
```
Comment (vydd) [inline src/utils.lisp:27]: cool

File: src/utils.lisp [hunk 3]
```diff
@@ -18,20 +18,10 @@
...
-                         list (cdr list))
-                finally (return (cons (nreverse acc) list))))))
-    (loop with acc = '()
-       while (or (not acc) (cdr list))
-       do (let ((split (split-n list group-length)))
-            (when (car split)
-              (setf acc (cons (car split) acc)))
-            (setf list (cdr split)))
-       finally (return (nreverse acc)))))
+  (loop with list = (copy-list list)
```
Comment (vydd) [inline src/utils.lisp:21]: cool

**Review by vydd — APPROVED** [review]

**Pull request merged**